array.c

```diff
@@ -5447,19 +5451,51 @@ rb_ary_concat(VALUE x, VALUE y)
 
 /*
  *  call-seq:
- *     ary * int     -> new_ary
- *     ary * str     -> new_string
+ *    array * n -> new_array
+ *    array * string_separator -> new_string
+ *
+ *  - Argument +n+, if given, must be an
+ *    {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
+ *  - Argument +string_separator+, if given, myst be a
+ *    {String-convertible object}[doc/implicit_conversion_rdoc.html#label-String-Convertible+Objects].
  *
- *  Repetition --- With a String argument, equivalent to
- *  <code>ary.join(str)</code>.
+ *  ---
  *
- *  Otherwise, returns a new array built by concatenating the +int+ copies of
```

Comment:
I think the older wording is more succinct and just as accurate, so I would keep it.    I would add your examples, though.

Reply:
I think I've done what you suggested.  Squawk if not.

Reply:
Due to the way you've moved things around, keeping the older wording would be replacing:

```
When argument +n+ is given, returns a new \Array built by concatenating +n+ copies of +self+.
If +n+ is positive, returns the concatenation of +n+ repetitions of +self+:
If +n+ is zero, returns an new empty \Array:
```

with: 

```
When argument +n+ is given, returns a new array built by concatenating the +int+ copies of +self+.
```

Reply:
One more time.

Reply:
Looks good, thank you.